compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -667,7 +671,7 @@ abstract class GenJSCode extends plugins.PluginComponent
             case lit: js.StringLiteral    => js.JSBracketSelect(selfRef, lit)
             case js.ComputedName(tree, _) => js.JSBracketSelect(selfRef, tree)
           }
-          js.Assign(select, jstpe.zeroOf(fdef.tpe))
+          js.Assign(select, jstpe.zeroOf(fdef.ftpe))
```

Comment:
This will naturally disappear once we rebase on top of the changes from the 0.6.x branch, yes. I can't remove it right now before we rebase, because `.tpe` simply does not typecheck.